src/MigrationFileSystem.php

```diff
@@ -17,16 +20,17 @@ public function __construct(private readonly string $doctrineMigrationsDir)
 
     public function newestMigrationFileName(): ?string
     {
-        $files = scandir($this->doctrineMigrationsDir, \SCANDIR_SORT_DESCENDING);
+        $finder = new Finder();
```

Comment:
@ismail1432 to do a quick test I added a requirement to the Symfony Finder component to find php files recursively.

In my use case I have a folder structure like this
```
migrations
    2022
       ...
    2023
        ...
        04
        05
            Version20230510155952.php
```

Reply:
It's a good improvement, less custom code less errors 👍 
Let's use the finder 🔥